src/Serializer/AbstractItemNormalizer.php

```diff
@@ -345,6 +348,21 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
         return $allowedAttributes;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    protected function isAllowedAttribute($classOrObject, $attribute, $format = null, array $context = [])
+    {
+        $options = $this->getFactoryOptions($context);
+        $propertyMetadata = $this->propertyMetadataFactory->create($context['resource_class'], $attribute, $options);
+        $security = $propertyMetadata->getAttribute('security');
+        if ($security && !$this->resourceAccessChecker->isGranted($attribute, $security)) {
+            return false;
+        }
+
+        return parent::isAllowedAttribute($classOrObject, $attribute, $format, $context);
```

Comment:
The parent method should be called first. No need to do extra work when we can return the value directly if it's already `false`. We only need to check the against the `security` attribute when the parent method returns `true`, since we're taking a blacklist approach here.